api/src/main/java/javax/mvc/locale/LocaleResolverContext.java

```diff
@@ -0,0 +1,73 @@
+/*
+ * DO NOT ALTER OR REMOVE COPYRIGHT NOTICES OR THIS HEADER.
+ *
+ * Copyright (c) 2015 Oracle and/or its affiliates. All rights reserved.
+ *
+ * The contents of this file are subject to the terms of either the GNU
+ * General Public License Version 2 only ("GPL") or the Common Development
+ * and Distribution License("CDDL") (collectively, the "License").  You
+ * may not use this file except in compliance with the License.  You can
+ * obtain a copy of the License at
+ * http://glassfish.java.net/public/CDDL+GPL_1_1.html
+ * or packager/legal/LICENSE.txt.  See the License for the specific
+ * language governing permissions and limitations under the License.
+ *
+ * When distributing the software, include this License Header Notice in each
+ * file and include the License file at packager/legal/LICENSE.txt.
+ *
+ * GPL Classpath Exception:
+ * Oracle designates this particular file as subject to the "Classpath"
+ * exception as provided by Oracle in the GPL Version 2 section of the License
+ * file that accompanied this code.
+ *
+ * Modifications:
+ * If applicable, add the following below the License Header, with the fields
+ * enclosed by brackets [] replaced by your own identifying information:
+ * "Portions Copyright [year] [name of copyright owner]"
+ *
+ * Contributor(s):
+ * If you wish your version of this file to be governed by only the CDDL or
+ * only the GPL Version 2, indicate your decision by adding "[Contributor]
+ * elects to include this software in this distribution under the [CDDL or GPL
+ * Version 2] license."  If you don't indicate a single choice of license, a
+ * recipient has the option to distribute your version of this file under
+ * either the CDDL, the GPL Version 2 or to extend the choice of license to
+ * its licensees as provided above.  However, if you add GPL Version 2 code
+ * and therefore, elected the GPL Version 2 license, then the option applies
+ * only if the new code is made subject to such option by the copyright
+ * holder.
+ */
+package javax.mvc.locale;
+
+import javax.servlet.http.HttpServletRequest;
+import javax.ws.rs.core.Configuration;
+
+/**
+ * <p>Contextual data used by a {@link javax.mvc.locale.LocaleResolver} to resolve the request
+ * locale. This includes the request from the Servlet container, the JAX-RS configuration,
+ * among other data.</p>
+ *
+ * @author Christian Kaltepoth
+ * @see javax.mvc.locale.LocaleResolver
+ * @see javax.servlet.http.HttpServletRequest
+ * @since 1.0
+ */
+public interface LocaleResolverContext {
+
+    /**
+     * Returns HTTP request object from the Servlet container.
+     *
+     * @return HTTP request object.
+     */
+    HttpServletRequest getRequest();
```

Comment:
Hey, thanks for the feedback. My comments:

**HttpHeaders for Accept-Language**

This would possible. BUT the header doesn't just contain a locale identifier but something like this:

    Accept-Language: de-DE,de;q=0.8,en-US;q=0.6,en;q=0.4

IIRC @spericas mentioned that there is some class called `VariantSelector` in Jersey which simplifies parsing that. For now I just used `HttpServletRequest#getLocale()` because it is simpler. But I agree that the default resolver could be improved later.

**LocaleResolverContext vs ContainerRequestContext**

I agree that `LocaleResolverContext` currently doesn't contain everything an implementation could need. But we can add more things to `LocaleResolverContext` very easily. That's the nice thing about these `*Context` interfaces. They can be extended without breaking compatibility. :)

To be honest, I don't think we should use `ContainerRequestContext` instead of `LocaleResolverContext` in the `LocaleResolver` API:

  * `ContainerRequestContext` has been defined for use with `ContainerRequestFilter` which is also mentioned in the corresponding javadocs. Using it for other SPIs may be confusing for users.
  * Using `LocaleResolverContext` gives us more flexibility because we can extend this interface in later versions with MVC-specific information without breaking anything.
  * Currently Ozark calls the `LocaleResolver` from a pre-matching container filter. This means that not all the information in the `ContainerRequestContext` is available. I would prefer to make it more explicit which information is available by just including things that are available in `LocaleResolverContext`.

However, I agree that we should add things like `UriInfo` and other useful objects to `LocaleResolverContext`. And we could even remove `HttpServletRequest` if we would like to get rid of servlet dependencies and prefer JAX-RS classes. But that's something we should discuss separately.